### PR TITLE
Add first prototype of Dockerator

### PR DESCRIPTION
**Pull request by drets**

### CODE REVIEW

File: dockerator.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+from docker.utils import kwargs_from_env
+
+
+def process(commands):
+
+    def get_image_name():
+        image_info = max(cli.images(), key=lambda i: i['Created'])
+        return image_info['RepoTags'][0].split(':')[0]
+
+    cli = Client(**kwargs_from_env())
```
Comment (st-kurilin) [inline dockerator.py]: why do need kwargs_from_env ?

File: dockerator.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+import yaml
+
+from docker.client import Client
+from docker.utils import kwargs_from_env
+
+
+def process(commands):
+
+    def get_image_name():
+        image_info = max(cli.images(), key=lambda i: i['Created'])
```
Comment (st-kurilin) [inline dockerator.py]: please provide comment about format

File: dockerator_tests.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+import subprocess as sp
+import urllib2
+import time
+
+
+class DockerSingleImage(unittest.TestCase):
+
+    def test_run_single_image(self):
+        cmd = ['python', 'dockerator.py', '-s', 'single_app/build.yaml']
+        sp.Popen(cmd, stdout=sp.PIPE).communicate()
```
Comment (st-kurilin) [inline dockerator_tests.py]: please provide some abstraction to make clear what is going on here ;)

File: dockerator.py (hunk 1 above)
Comment (drets) [inline dockerator.py]: Yep, I am planning change it to base_url = where the Docker server is hosted.

File: dockerator_tests.py (hunk 3 above)
Comment (st-kurilin) [inline dockerator_tests.py]: something like 

run_dockerator('path_to_image')
http_get(http://192.168.59.103:1111/, timeout_ms = 1000)  == 'Hello world'

**Comment by st-kurilin** [comment]
Looks nice for me. What did you mean by 'single_app'?

File: dockerator_tests.py (hunk 3 above)
Comment (drets) [inline dockerator_tests.py]: Yep, I am going to refactor it.

**Comment by drets** [comment]
I meant the simple web python app.

File: dockerator.py (hunk 2 above)
Comment (drets) [inline dockerator.py]: Sure, thanks!

File: dockerator_tests.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+        def http_get(url, timeout):
+            start_time = time.time()
+            while time.time() - start_time < timeout:
+                try:
+                    return urllib2.urlopen(url).read()
+                except:
+                    continue
+
+        run_dockerator('tcp://192.168.59.103:2375',
```
Comment (st-kurilin) [inline dockerator_tests.py:24]: nice :+1:

File: dockerator_tests.py [hunk 5]
```diff
@@ -0,0 +1,30 @@
+import unittest
+import subprocess as sp
+import urllib2
+import time
+
+
+class DockerSingleImage(unittest.TestCase):
+
+    def test_run_single_image(self):
```
Comment (st-kurilin) [inline dockerator_tests.py]: Perhaps test_run_simple_web_app?

File: dockerator.py [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+from docker.client import Client
+
+
+def process(base_url, commands):
+
+    def get_image_name():
+        # image_info is dict.
+        # RepoTags key has ['image_name:tag', ...] value.
+        image_info = max(cli.images(), key=lambda i: i['Created'])
```
Comment (st-kurilin) [inline dockerator.py]: could you please pass explicitly `cli` and give better name to it.

**Comment by st-kurilin** [comment]
Looks really nice to me ;) Did you try run pylint?

**Comment by drets** [comment]
Thanks. I use flake8.

**Comment by st-kurilin** [comment]
ok. 
you can merge it)

**Pull request merged**